lib/endpoints/class-wp-rest-comments-controller.php

```diff
@@ -177,6 +185,15 @@ public function create_item( $request ) {
 		$prepared_comment['comment_agent'] = '';
 		$prepared_comment['comment_approved'] = wp_allow_comment( $prepared_comment );
 
+		// Check author name and email if required.
+		if ( get_option( 'require_name_email' ) && ! isset( $user ) ) {
+			if ( empty( $prepared_comment['comment_author_email'] ) || empty( $prepared_comment['comment_author'] ) ) {
+				return new WP_Error( 'rest_require_valid_comment', __( 'Required fields (name, email) missing.' ), array( 'status' => 400 ) );
+			} elseif ( ! is_email( $prepared_comment['comment_author_email'] ) ) {
```

Comment:
The `is_email` check is already done in `WP_REST_Controller->validate_schema_property` no need to duplicate it here.